WcaOnRails/spec/mailers/previews/delegate_status_change_mailer_preview.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+# Preview all emails at http://localhost:3000/rails/mailers/wca_id_claim_mailer
+class DelegateStatusChangeMailerPreview < ActionMailer::Preview
+  def notify_board_and_wqac_of_delegate_status_change
+    ActiveRecord::Base.transaction do
+      user_whose_delegate_status_changes = User.where.not(delegate_status: nil).first
```

Comment:
As you pointed out in your original post, this is a bit weird because this user's delegate status has not actually changed. I think one fix would be to write it like this instead:

```ruby
delegate_being_promoted = User.where(delegate_status: "candidate_delegate").first
delegate_being_promoted.delegate_status = "delegate"
...
```

This should set up a delegate who is in the process of being promoted, but won't actually persist that change to the database.

Reply:
This sounds like a good approach, I will make this change too :)